File: 프로그래머스/lv2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int answer = 0;

        Queue<Long> q1 = new LinkedList<>();
        Queue<Long> q2 = new LinkedList<>();

        long sum1 = 0;
        long sum2 = 0;

        long totalCount = (queue1.length + queue2.length) * 2;

        for(int i = 0; i < queue1.length; i++){
            q1.add((long)queue1[i]);
            sum1 += queue1[i];
            q2.add((long)queue2[i]);
            sum2 += queue2[i];
        }


       while(sum1 != sum2){
        answer++;

        long poll;
        if (sum1 > sum2) {
            poll = q1.poll();
            sum1 -= poll;
            sum2 += poll;
            q2.offer(poll);
        }
         else{
            poll = q2.poll();
            sum1 += poll;
            sum2 -= poll;
            q1.offer(poll);
        }

        if (answer > totalCount) {
            return -1;
        }
        
       }

        return answer;
    }
}

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from queue import Queue

def solution(queue1, queue2):
    answer = 0
    q1 = Queue()
    q2 = Queue()
    total_q1, total_q2 = 0, 0

    for i in range(len(queue1)):
        num1 = queue1[i]
        num2 = queue2[i]
        q1.put(num1)
        q2.put(num2)
        total_q1 += num1
        total_q2 += num2

    max_operations = len(queue1) * 3  

    while total_q1 != total_q2:
        if answer > max_operations:
            return -1

        if total_q1 > total_q2:
            num = q1.get()
            q2.put(num)
            total_q1 -= num
            total_q2 += num
        else:
            num = q2.get()
            q1.put(num)
            total_q2 -= num
            total_q1 += num

        answer += 1

    return answer
